apputils/containerized.py

```diff
@@ -18,107 +20,117 @@ def __init__(self):
         self.scans_to_s3_ver = os.getenv('SCANS_TO_S3_VERSION', 'latest')
         self.events_to_s3_ver = os.getenv('EVENTS_TO_S3_VERSION', 'latest')
 
+    def generic_container_launch_attached(self, image, env_vars, env_expand,
+                                          read_only=True):
+        """Launch a container, return the output from container's STDOUT.
+
+
+        Args:
+            image(str): Full image path. For example:
+                ``docker.io/halotools/halocelery:latest``
+            env_vars(dict): Environment variables in a dictionary where key
+                and value are ``{'key1': 'value1', 'key2': 'value2'}``
+            env_expand(dict): Environment variable names (as seen inside the
+                container) are keys and the values are to be expanded from the
+                environment variables available to the running celery worker.
+                For instance, if the container needs an environment variable
+                named ``FOO`` to contain the value of the celery worker's
+                environment variable named ``BAR``, this argument would take
+                the form of ``{'FOO': 'BAR'}``. If the celery worker has an
+                environment variable named ``BAR``, which contains the value
+                ``BAZ``, and arg:env_expanded is ``{'FOO': 'BAR'}``, the
+                environment inside the running container will contain an
+                environment variable named ``FOO`` with the value ``BAZ``.
+                In short, the keys are persisted into the container and the
+                values are replaced with the corresponding environment
+                variables, according to the environment running celery.
+            read_only(bool): Set to ``False`` to allow writeable access to
+                filesystem inside container.
+
+        """
+        container_name = self.generate_random_name()
+        env_vars = self.expand_and_update_env_vars(env_vars, env_expand)
+        print("ContainerLauncher: Launching %s from %s" % (container_name,
+                                                           image))
+        result = self.client.containers.run(image, name=container_name,
+                                            detach=False,
+                                            mem_limit=self.mem_limit,
+                                            environment=env_vars)
+        self.client.containers.get(container_name).remove()
+        return result
+
+    @classmethod
+    def generate_random_name(cls):
```

Comment:
Consider using python module 'uuid' to generate a true unique name.
Make a UUID based on the host ID and current time:

import uuid

id = str(uuid.uuid1())
print id

output: '14b83238-5534-11e8-9b1d-f45c8990a101'

Reply:
Great idea, I'll do this instead.